src/features/SendAssets/components/SendAssetsForm.tsx

```diff
@@ -361,432 +363,383 @@ const SendAssetsForm = ({ txConfig, onComplete }: IStepComponentProps) => {
+          accounts={accountsWithAsset}
+          onSelect={(account: StoreAccount) => {
+            setFieldValue('account', account); //if this gets deleted, it no longer shows as selected on interface, would like to set only object keys that are needed instead of full object
+            handleGasEstimate();
+          }}
+          asset={values.asset}
+        />
+      </fieldset>
+      <fieldset className="SendAssetsForm-fieldset">
+        <label htmlFor="address" className="input-group-header">
+          {translate('X_RECIPIENT')}
+        </label>
+        <ContactLookupField
+          name="address"
+          value={values.address}
+          error={errors && touched.address && errors.address && (errors.address as ErrorObject)}
+          network={values.network}
+          isResolvingName={isResolvingName}
+          onBlur={handleGasEstimate}
+          setIsResolvingDomain={setIsResolvingDomain}
+          setFieldValue={setFieldValue}
+          setFieldTouched={setFieldTouched}
+          setFieldError={setFieldError}
+        />
+      </fieldset>
+      {/* Amount */}
+      <fieldset className="SendAssetsForm-fieldset">
+        <label htmlFor="amount" className="input-group-header label-with-action">
+          <div>{translate('SEND_ASSETS_AMOUNT_LABEL')}</div>
+          <div className="label-action" onClick={setAmountFieldToAssetMax}>
+            {translateRaw('SEND_ASSETS_AMOUNT_LABEL_ACTION').toLowerCase()}
+          </div>
+        </label>
+        <>
+          <AmountInput
+            name="amount"
+            onChange={(e) => {
+              setFieldValue('amount', e.target.value);
+            }}
+            asset={values.asset}
+            value={values.amount}
+            onBlur={() => {
               setFieldTouched('amount');
-              setIsEstimatingGasLimit(false);
-            } else {
-              return;
-            }
-          };
-
-          const setAmountFieldToAssetMax = () => {
-            const account = getAccount(values.account);
-            if (values.asset && account && baseAsset) {
-              const accountBalance = getAccountBalance(account, values.asset).toString();
-              const isERC20 = isERC20Tx(values.asset);
-              const balance = fromTokenBase(new BN(accountBalance), values.asset.decimal);
-              const gasPrice = values.advancedTransaction
-                ? values.gasPriceField
-                : values.gasPriceSlider;
-              const amount = isERC20 // subtract gas cost from balance when sending a base asset
-                ? balance
-                : baseToConvertedUnit(
-                    new BN(convertedToBaseUnit(balance.toString(), DEFAULT_ASSET_DECIMAL))
-                      .sub(gasStringsToMaxGasBN(gasPrice, values.gasLimitField))
-                      .toString(),
-                    DEFAULT_ASSET_DECIMAL
-                  );
-              setFieldValue('amount', amount);
               handleGasEstimate();
```

Comment:
Let effect handle this.

Reply:
This is for amount, so I wouldn't recommend to use an effect.

Reply:
After second thought the amount shouldn't change the gasEstimate.
Let's move to effect and useThrottle https://github.com/streamich/react-use